packages/design-system/src/components/IdleTimeout/IdleTimeout.tsx

```diff
@@ -77,70 +78,148 @@ const defaultMessageFormatter = (timeTilTimeout: number): React.ReactNode => {
   );
 };
 
+// names for local storage variables
+const timeoutCookieName = 'CMS_DS_TIMEOUT';
+const timeoutWarningCookieName = 'CMS_DS_WARNING';
+
 const IdleTimeout = ({
-  closeDialogText = 'Close',
   continueSessionText = 'Continue session',
   heading = 'Are you still there?',
   endSessionButtonText = 'Logout',
+  endSessionRedirectUrl = '/logout',
   formatMessage = defaultMessageFormatter,
-  onClose,
   onSessionContinue,
   onSessionForcedEnd,
   onTimeout,
   showSessionEndButton = false,
   timeToTimeout,
   timeToWarning = 5,
-}: IdleTimeoutProps) => {
-  const handleContinueSession = () => {
-    // reset countdown timer
-    if (onSessionContinue) {
-      onSessionContinue();
+}: IdleTimeoutProps): JSX.Element => {
+  if (timeToWarning > timeToTimeout) {
+    console.error(
+      'Error in TimeoutManager component. `timeToWarning` is greater or equal to `timeToTimeout`'
+    );
+  }
+  // convert minutes to milliseconds
+  const msToTimeout = timeToTimeout * 60000;
+  const msToWarning = (timeToTimeout - timeToWarning) * 60000;
+  const [checkStatusTime, setCheckStatusTime] = useState<number>(null);
+  const [warningIntervalId, setWarningIntervalId] = useState<NodeJS.Timeout>(null);
```

Comment:
> I found this (https://stackoverflow.com/a/56239226) when searching for NodeJS.Timeout, seems like it may be better typing for this for platform independence?

From a different PR -- wanted to capture it here

Reply:
I think it would be good to use the return type, since it adapts to any environment and always stays accurate. I'm pretty sure I've actually used that method before. Could have even seen the same SO answer.

Reply:
This is no longer needed since we aren't keeping track of any interval ids with the refactor!
But I will keep this in mind for the future. TIL 🌈